nion/swift/Inspector.py

```diff
@@ -588,109 +588,97 @@ def __property_changed_from_graphic(self, name: str) -> None:
             self.value = getattr(self.__graphic, self.__property_name)
 
 
+class InfoInspectorHandler(Declarative.Handler):
+    def __init__(self, document_controller: DocumentController.DocumentController, display_item: DisplayItem.DisplayItem):
+        super().__init__()
+        self._display_item = display_item
+        self._title_model = Model.PropertyChangedPropertyModel[str](display_item, "title")
+        self._placeholder_title_model = Model.PropertyChangedPropertyModel[str](display_item, "placeholder_title")
+        self._caption_model = Model.PropertyChangedPropertyModel[str](display_item, "caption")
+        self._editable_caption_model = Model.PropertyModel[str](display_item.caption)
```

Comment:
To make editing undo-able, `_title_model` and either `_caption_model` or `_editable_caption_model` can use `DisplayItemPropertyCommandModel`. Also see note below.

Reply:
I have done the _caption_model, but the other won't work:
_title_model - I get a conflict making a DisplayPanel.ChangeDisplayCommand because it already has a (different) title input giving me the error "nion.swift.DisplayPanel.ChangeDisplayCommand() got multiple values for keyword argument 'title'"
_editable_caption_model - I was trying to maintain the existing behaviour where it does not get pushed to the display model until the save button is pressed - this would change that.

Reply:
Yes I missed this comment... looking at it now.

Reply:
Ok here's what's going on... it not pretty. There are two types of "properties" on the display item. One is the direct properties - things like the `title` and `caption`. The other properties are display type specific and include things like `y_min` and `left_channel` for line plots, which are stored within a sub-structure of the display item named "display properties".  Setting aside the design of this (needs work!), within this context, the `DisplayItemPropertyCommandModel` is mis-named and should be renamed to`DisplayItemDisplayPropertyCommandModel` (also update in `LinePlotDisplaySectionHandler`). Crucially, that class uses `DisplayPanel.ChangeDisplayCommand` to call `set_display_property` on the display item in an undoable way. Then you'll need to add a new class named `DisplayItemCommandModel` which uses `ChangeDisplayItemPropertyCommand` to set a property directly on the display item. `title` and `caption` should use the _new_ `DisplayItemCommandModel`.